client/files.go

```diff
@@ -78,14 +78,16 @@ type fileResult struct {
 }
 
 type errorResult struct {
-	Message string `json:"message"`
+	Message string      `json:"message"`
+	Kind    string      `json:"kind,omitempty"`
+	Value   interface{} `json:"value,omitempty"`
 }
 
 // MakeDir creates a directory or directory tree.
 func (client *Client) MakeDir(opts *MakeDirOptions) error {
 	var permissions string
-	if opts.Permissions != nil {
-		permissions = fmt.Sprintf("%o", *opts.Permissions)
+	if opts.Permissions != 0 {
+		permissions = fmt.Sprintf("%3o", opts.Permissions)
```

Comment:
Per my original comment, this needs to be `%03o`, so it's zero-padded rather than space-padded. Otherwise you get:

```
$ pebble mkdir /home/ben/w/pebble/foo -m 25
error: permissions must be a 3-digit octal string, got " 25"
exit status 1
```

We should probably add a test for a 1 or 2 digit mode to catch this.